### PR TITLE
Checkout: Thank you: Redesign

### LINKED ISSUE #2484: Post Checkout: update post-purchase thank you page design
Raised by @mikeshelton1503 

As part of the No Credit Card free trials we are testing a new version of the Thank You page to increase engagement (#1124). Pending the results of that test we'll want to update all Thanks Page if the new design wins.

For reference, this is what it looks like:
image

The layout and design of the page will remain the same for all purchases but as we do now we will change the content per each type of purchase.

The different version we have:
 - Premium purchase
 - Business purchase
 - Domain registration
 - Domain mapping
 - Google apps
 - Site redirect
 - Generic

Screenshots of the current pages: https://cloudup.com/cTfdvTFheu4

Mobile version:

### Mobile

thanks-page-premium-v3-mobile

### PR DESCRIPTION
**Pull request by scruffian**
This pull request adds the designs from #2484. We will do the copy changes in separate PRs

The original mockups look like this.
Screenshot
Screenshot

#### Testing instructions

1. Run `git checkout update/checkout-thank-you-page-redesign` and start your server
2. Purchase the a product and assert that the thank you page:
3. Has the correct placeholders on mobile
3. Has the correct placeholders on desktop
3. Has the correct design on mobile
3. Has the correct design on desktop

#### Additional notes

It will be worth purchasing a few different products to see how the page looks with different content.

#### Reviews

- [x] Code
- [ ] Product

### CODE REVIEW

File: client/my-sites/upgrades/checkout-thank-you/style.scss [hunk 1]
```diff
@@ -30,47 +30,30 @@
...
 
-	@include breakpoint( ">660px" ) {
-		padding: 20px;
-		border-bottom: none;
-
-		> h3 {
-			display: inline;
-			margin-right: 15px;
-		}
+.checkout-thank-you__support_heading {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/style.scss]: The component name in this class needs to be kebab-case.

**Comment by drewblaisdell** [comment]
I added a commit to remove `locale` as a prop of the product-specific components (none of them used it), and removed the locale slug from the links in `supportRelatedMessages` (@stephanethomas has pointed  out that we're better off omitting these).

File: client/my-sites/upgrades/checkout-thank-you/header/index.jsx [hunk 2]
```diff
@@ -50,16 +62,21 @@ const CheckoutThankYouHeader = React.createClass( {
...
-				<span className="checkout-thank-you-header__icon"/>
-				<h1 className="checkout-thank-you-header__heading">
-					{ this.renderHeading() }
-				</h1>
-				<h2 className="checkout-thank-you-header__text">
-					{ this.renderText() }
-				</h2>
-			</div>
+			<CompactCard className={ classNames( classes ) }>
+				<div className="checkout-thank-you-header-content">
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/header/index.jsx]: This class should be named `checkout-thank-you-header__content`.

File: client/my-sites/upgrades/checkout-thank-you/index.jsx [hunk 3]
```diff
@@ -193,55 +200,50 @@ var CheckoutThankYou = React.createClass( {
...
-					isFreeTrial: this.freeTrialWasPurchased(),
-					locale: i18n.getLocaleSlug(),
-					domain: domain
-				} ) }
+				<CompactCard className="checkout-thank-you__features-header">
+					{ this.translate( "Get started with your site's new features" ) }
+				</CompactCard>
+
+				<div className="checkout-thank-you__purchase-details-list">
+					{ React.createElement( componentClass, {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: Not introduced by you, but we can get rid of the `React.createElement` call here and just use JSX:

```js
<ComponentClass
  selectedSite={ selectedSite }
  isFreeTrial={ this.freeTrialWasPurchased() }
  domain={ domain } />
```

**Comment by drewblaisdell** [comment]
Looking good so far. :+1:

**Comment by breezyskies** [comment]
I've added a commit to clean up some of the padding and font styles. Also a few notes:

- Can we use the full product name (e.g. "WordPress.com Premium" instead of just "Premium")?
- I'm seeing free trial content in the heading after paying for Premium (not sure if this is still WIP).

@mikeshelton1503 As discussed in Slack, I tightened up the padding a bit to account for less content in this version than in your mockups. We can readdress after the new content is in. Feel free to add additional tweaks if you spot any.

**Comment by drewblaisdell** [comment]
> Can we use the full product name (e.g. "WordPress.com Premium" instead of just "Premium")?

I included this change in #3548 (and the associated patch). We can pull it into this PR if that makes more sense (but both need to change `CheckoutThankYouHeader`).

**Comment by scruffian** [comment]
> I included this change in #3548 (and the associated patch). We can pull it into this PR if that makes more sense (but both need to change CheckoutThankYouHeader).

Lets keep this PR as design changes only.

**Comment by scruffian** [comment]
> I'm seeing free trial content in the heading after paying for Premium (not sure if this is still WIP).

I can't reproduce this. :/

**Comment by scruffian** [comment]
@breezyskies - I'm not sure how to move forward with this. We might need to have a discussion about some of these points to pin down the details....

File: client/my-sites/upgrades/checkout-thank-you/index.jsx [hunk 4]
```diff
@@ -193,55 +198,50 @@ var CheckoutThankYou = React.createClass( {
...
-				} ) }
+				<div className="checkout-thank-you__features-header">
+					{ this.translate( "Get started with your site's new features" ) }
+				</div>
+
+				<div className="checkout-thank-you__purchase-details-list">
+					{ React.createElement( componentClass, {
+						selectedSite: selectedSite,
+						isFreeTrial: this.freeTrialWasPurchased(),
+						domain: domain
```
Comment (stephanethomas) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: You could use a shorthand property here.

**Comment by stephanethomas** [comment]
The code looks good so far.

**Comment by mikeshelton1503** [comment]
Woohoo, glad to see this getting implemented! :smile: I noticed a few things:

- The icon in the header area is off-center a bit:
image

- The headline at the top of the page should be semibold, looks like bold right now.
- The header area could probably use even more top and bottom padding. Measuring the comp, looks like `90px` should do it, on large screens.
- The trophy icon at the top should be `darken( $gray, 20% )`
- Below 660px the white circle behind the trophy icon is a little too big. The icon is the right size though. The circle should be `90px`.

**Comment by breezyskies** [comment]
Pushed several CSS fixes for the card issue, borders, and moved the header to flexbox. I still need to take a look at the individual feature icons, but won't get to that until later today.

File: client/my-sites/upgrades/checkout-thank-you/index.jsx (hunk 4 above)
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: @stephanethomas I don't think we want to use ES6 syntax in modules unless the entire module uses that syntax. We should use JSX here, though.

File: client/my-sites/upgrades/checkout-thank-you/index.jsx [hunk 5]
```diff
@@ -49,7 +48,7 @@ function getPurchases( props ) {
 	return props.receipt.data.purchases;
 }
 
-var CheckoutThankYou = React.createClass( {
+const CheckoutThankYou = React.createClass( {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: @scruffian I don't think we should be using `const` here until the entire module uses ES6.

**Comment by scruffian** [comment]
I also noticed that the placeholders on this are off:
screen shot 2016-02-25 at 18 00 15

File: client/my-sites/upgrades/checkout-thank-you/index.jsx (hunk 4 above)
Comment (stephanethomas) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: I agree but if we don't start somewhere we're going to implement the new `Thank You` page and it will still be old Javascript :).

**Comment by scruffian** [comment]
> I also noticed that the placeholders on this are off:

I added c9a889f to address this.

**Comment by scruffian** [comment]
> I also noticed that the placeholders on this are off:

I added c9a889f to address this.

File: client/my-sites/upgrades/checkout-thank-you/index.jsx (hunk 5 above)
Comment (scruffian) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: Maybe we can consider moving to ES6 gradually in this case since we will have so many PR touching the code at the same time? Or is that a bad idea?

**Comment by breezyskies** [comment]
Pushed one last fix for vertically aligning the Noticons in the feature block in https://github.com/Automattic/wp-calypso/commit/496b267d8920a442159f9bd5976a5a5600f4e1b0. We should switch to flexbox for this when we move to Gridicons, but that doesn't play very nice with `:before`, so this will work for now.

**Comment by scruffian** [comment]
Thanks @breezyskies

**Comment by drewblaisdell** [comment]
We should probably squash [this](https://github.com/Automattic/wp-calypso/commit/13066820328bc992c083d79187bb76e771a50380) before merging.

File: client/my-sites/upgrades/checkout-thank-you/header/index.jsx [hunk 6]
```diff
@@ -42,13 +43,27 @@ const CheckoutThankYouHeader = React.createClass( {
 		}
 
 		if ( this.props.primaryPurchase ) {
-			return this.translate(
-				"You will receive an email confirmation shortly for your purchase of %(productName)s. What's next?", {
+			if ( this.props.isFreeTrial ) {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/header/index.jsx]: `primaryPurchase` has a free trial property, so we can use `isFreeTrial( this.props.primaryPurchase )` here and obviate that prop entirely.

File: client/my-sites/upgrades/checkout-thank-you/index.jsx (hunk 5 above)
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: It is a large component, so it might make sense as its own PR. We only have one other PR for this component open at the moment and the merge conflicts wouldn't be that bad if this PR updated `CheckoutThankYou` to use ES6.

File: client/my-sites/upgrades/checkout-thank-you/header/index.jsx [hunk 7]
```diff
@@ -42,13 +43,27 @@ const CheckoutThankYouHeader = React.createClass( {
...
 					args: {
-						productName: this.props.primaryPurchase.productName
+						productName: this.props.productName
+					},
+					components: {
+						strong: <strong/>
+					}
+				} );
+			} else {
+				return this.translate(
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/header/index.jsx]: We have a [linting rule](https://github.com/Automattic/wp-calypso/blob/master/.eslintrc.js#L34) for unnecessary else clauses like this.

**Comment by drewblaisdell** [comment]
If you're cool with my commit, this looks good to me.

File: client/my-sites/upgrades/checkout-thank-you/style.scss [hunk 8]
```diff
@@ -11,165 +11,211 @@
...
+}
 
-		> p {
-			display: inline;
-		}
+.checkout-thank-you__message {
+	padding-bottom: 20px;
+}
+
+.checkout-thank-you__receipt-icon {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/style.scss]: This is dead CSS.

File: client/my-sites/upgrades/checkout-thank-you/style.scss [hunk 9]
```diff
@@ -11,165 +11,211 @@
...
+.checkout-thank-you__support-related-messages {
+	color: darken( $gray, 10% );
+	font-size: 16px;
+	margin: 8px 0 0 0;
+}
 
-		> p {
-			display: inline;
-		}
+.checkout-thank-you__message {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/style.scss]: I believe this is also dead CSS.

File: client/my-sites/upgrades/checkout-thank-you/style.scss [hunk 10]
```diff
@@ -11,165 +11,211 @@
...
-	@include breakpoint( "<660px" ) {
-		display: inline-block;
-		margin: 0 0 10px 0;
+	@include breakpoint( ">660px" ) {
+		margin: 0 24px;
+		padding-bottom: 48px;
+	}
+}
+
+.checkout-thank-you__product-related-messages {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/style.scss]: This is also dead CSS.

**Comment by scruffian** [comment]
Changes look good, thanks @drewblaisdell.

**Comment by fabianapsimoes** [comment]
Tiny nitpicks

- "Thank you for your purchase" should be bold, according to the mockup. It looks like this now:

screenshot-calypso localhost 3000 2016-02-26 11-28-42

- Some gridicons don't look quite centered (I guess that's the case for the trophy in the main banner as well):

screenshot-calypso localhost 3000 2016-02-26 11-34-51

I tested domain products, Google Apps, Premium and Business. They all look fine.

**Comment by scruffian** [comment]
@fabianapsimoes I think we'll have to wait for #3578 to fix the icon alignment.

@breezyskies the header text looks different to the mockup because we are now using system fonts. Are you still happy with using a weight of `400` or should we increase to `600`?

**Comment by breezyskies** [comment]
> @breezyskies the header text looks different to the mockup because we are now using system fonts. Are you still happy with using a weight of 400 or should we increase to 600?

Let's keep `400`.

> Some gridicons don't look quite centered (I guess that's the case for the trophy in the main banner as well)

Could use a second pair of eyes on the icons CSS. Both @mikeshelton1503 and I looked at that yesterday and we weren't sure why that was happening.

**Comment by scruffian** [comment]
@breezyskies, given the changes coming #3578, lets merge this as is, and make further adjustments there.

**Comment by breezyskies** [comment]
:+1:
